recipes/gdb/meta.yaml

```diff
@@ -0,0 +1,53 @@
+{% set name = "gdb" %}
+{% set version = "7.11" %}
+{% set sha256 = "7a434116cb630d77bb40776e8f5d3937bed11dea56bafebb4d2bc5dd389fe5c1" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: gdb-{{ version }}.tar.xz
+  url: http://ftp.gnu.org/gnu/gdb/gdb-{{ version }}.tar.xz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win or osx]
+
+requirements:
+  build:
+    - python
+    - ncurses ==5.9
+    - texinfo
+    - toolchain
+    - xz 5.2.*
+  run:
+    - python
+    - ncurses >=5.9,<6.0
+    - xz >=5.2,<6.0
```

Comment:
Thanks for the interest.
These compiled libraries seem to follow semantic version rules. So:
- they don't add any symbols when incrementing patch versions. That is why the recipe don't care exactly which version of ```xz``` is building against. Any patch version is equivalent in terms of ABI
- minor versions can add symbols (but not modify existing ones). That's why we need to pin ```xz``` and ```ncurses``` minor versions during build, but can allow to run with larger minor versions (they can only add things afterall). 
- and major, well, they can just break everything, so the package should be totally rebuilt

For ```xz``` you can find an ABI report here: https://abi-laboratory.pro/tracker/timeline/xz/.
Note that all versions are backward compatible, but from ```5.0.8``` to ```5.2.0``` they added 5 new symbols.

For ```ncurses``` I could not find such a report, but the release notes for 6.0 can give an idea: https://www.gnu.org/software/ncurses/

We could of course use the same pinning for running as we used for building here, but that would be too restrictive in terms of where the package could be installed and we would much more likely run into conflicts.